compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -380,22 +380,34 @@ fn negative_impl_exists<'cx, 'tcx>(
 
 #[instrument(level = "debug", skip(infcx))]
 fn resolve_negative_obligation<'cx, 'tcx>(
-    infcx: &InferCtxt<'cx, 'tcx>,
-    param_env: ty::ParamEnv<'tcx>,
+    infcx: InferCtxt<'cx, 'tcx>,
     o: &PredicateObligation<'tcx>,
+    body_def_id: DefId,
 ) -> bool {
     let tcx = infcx.tcx;
 
     let Some(o) = o.flip_polarity(tcx) else {
         return false;
     };
 
-    let errors = super::fully_solve_obligation(infcx, o);
-    if !errors.is_empty() {
+    let param_env = o.param_env;
+    if !super::fully_solve_obligation(&infcx, o).is_empty() {
         return false;
     }
 
-    let outlives_env = OutlivesEnvironment::new(param_env);
+    let outlives_env = if let Some(body_def_id) = body_def_id.as_local() {
```

Comment:
please do not change the used implied bounds based on whether the impl is from the current crate, at least not without a comment explaining the reasoning why.

You can use the `CRATE_HIR_ID` for obligations from foreign crates

Reply:
@lcnr fixed in https://github.com/rust-lang/rust/pull/100888/commits/4da14ef50eb697298f36d41f91cb02dc800275bf